.ci/setup.sh

```diff
@@ -20,7 +20,7 @@ if [[ $OS_NAME == "macos" ]]; then
         # Check https://github.com/actions/runner-images/tree/main/images/macos for available
         # versions of Xcode
         sudo xcode-select -s /Applications/Xcode_14.3.1.app/Contents/Developer || exit 1
-        brew install gcc
+        brew install 'gcc@12'
```

Comment:
I think we should continue using the most recent gcc here.

`brew install gcc` should install gcc@14 right now according to the docs:

> gcc
> Install command:
> brew install gcc
> Also known as: gcc@14, gfortran
https://formulae.brew.sh/formula/gcc#default

And seems it does:
```
Warning: gcc 14.1.0_2 is already installed and up-to-date.
To reinstall 14.1.0_2, run:
  brew reinstall gcc
```

Can you please revert this particular change and see whether we will get that error again. I guess it ether was temporal or comes from another place in our CI config.
```
CMake Error at /usr/local/Cellar/cmake/3.30.2/share/cmake/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  gcc-11.
Call Stack (most recent call first):
  CMakeLists.txt:36 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```